src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffoldSteps/DotnetNewScaffolderStep.cs

```diff
@@ -54,13 +55,21 @@ public override Task<bool> ExecuteAsync(ScaffolderContext context, CancellationT
 
     private bool InvokeDotnetNew(DotnetNewStepSettings stepSettings)
     {
-        var projectBasePath = Path.GetDirectoryName(stepSettings.Project);
-        //using the project name from the csproj path as namespace currently.
-        //to evaluate the correct namespace is a lot of overhead for a simple 'dotnet new' operation
-        //TODO maybe change this?
+        var outputDirectory = Path.GetDirectoryName(stepSettings.Project);
+        //invoking a command with a specific output folder, if not, use the default output folder (project root)
+        if (!string.IsNullOrEmpty(outputDirectory) &&
+            OutputFolders.TryGetValue(stepSettings.CommandName, out var folderName))
+        {
+            outputDirectory = Path.Combine(outputDirectory, folderName);
+            if (!_fileSystem.DirectoryExists(outputDirectory))
+            {
+                _fileSystem.CreateDirectory(outputDirectory);
```

Comment:
We should probably add a CreateDirectoryIfNotExists helper method on there since we do this a few times and it makes the code a little verbose when it doesn't need to be. Don't have to do it now if you don't want to, just a suggestion.